src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1262,6 +1303,21 @@ void SolverInterfaceImpl::readScalarData(
   PRECICE_TRACE(dataID, valueIndex, value);
   PRECICE_CHECK(_state != State::Finalized, "readScalarData(...) cannot be called after finalize().");
   PRECICE_REQUIRE_DATA_READ(dataID);
+  double dt = _couplingScheme->getThisTimeWindowRemainder(); // samples at end of time window
+  return readScalarData(dataID, valueIndex, dt, value);
+}
+
+void SolverInterfaceImpl::readScalarData(
+    int     dataID,
+    int     valueIndex,
+    double  dt,
+    double &value) const
+{
+  PRECICE_TRACE(dataID, valueIndex, value);
+  PRECICE_CHECK(_state != State::Finalized, "readScalarData(...) cannot be called after finalize().");
+  PRECICE_CHECK(dt <= _couplingScheme->getThisTimeWindowRemainder(), "readScalarData(...) cannot sample data outside of current time window.");
+  PRECICE_ASSERT(dt == _couplingScheme->getThisTimeWindowRemainder()); // @todo only sampling at end of window currently works.
```

Comment:
?

Reply:
In this PR currently only the API is extended. The actual functionality is still in https://github.com/BenjaminRodenberg/precice/pull/1. Then the todo will also vanish:

See https://github.com/BenjaminRodenberg/precice/pull/1/files#diff-cf9df0e4694b35c558b21d42de07db93a0bf82ec41a311e54350a5fafe1306d6L1319